docs/src/_reference/command-line-interface.md

```diff
@@ -120,6 +120,55 @@ meltano add extractor tap-spotify --no-install
 
 The `add` command does not run relative to a [Meltano Environment](https://docs.meltano.com/concepts/environments). The `--environment` flag and [`default_environment` setting](https://docs.meltano.com/concepts/environments#default-environments) in your `meltano.yml` file will be ignored if set.
 
+## `compile`
+
+### How to use
+
+Generally, the `compile` command need not be executed manually, as it will be run automatically by Meltano as-needed.
+
+If you wish to manually compile a manifest file for each environment (including the no-environment manifest file), the compile command can be used like so:
+
+```bash
+meltano compile
+```
+
+To compile a manifest file for a specific environment:
+
+```bash
+meltano --environment <environment name> compile
+```
+
+To save the manifest JSON files to a specific directory:
+
+```bash
+meltano compile --directory /some/directory/path
+```
+
+By default, the manifest files are saved to `${MELTANO_SYS_DIR_ROOT}/manifests`, which defaults to `${MELTANO_PROJECT_ROOT}/.meltano/manifests`.
+
+Use the `--indent` CLI option to control the indentation in the manifest JSON files:
+
+```bash
+# Use 2 spaces of indentation instead of the default 4
+meltano compile --indent 2
+
+# Only use newlines
+meltano compile --indent 0
+
+# Remove all non-essential whitespace
+meltano compile --indent -1
+```
+
+### Using `compile` with Environments
+
+The `compile` command can accept the `--environment` flag to target a specific [Meltano Environment](https://docs.meltano.com/concepts/environments). However, the [`default_environment` setting](https://docs.meltano.com/concepts/environments#default-environments) in your `meltano.yml` file will be ignored.
+
+When an environment is specified, only the manifest JSON file for that environment will be compiled.
+
+When no environment is specified, a manifest JSON file for each environment is compiled, including `meltano-manifest.json`, which is the manifest file for the project when no environment is active.
+
+There is no way to *only* compile the no-environment manifest JSON file, i.e. `meltano-manifest.json`. It is only compiled when the `compile` command is run on all environments.
```

Comment:
Could the `--no-environment` flag be used to compile the single file?

Reply:
Yes, if I add another `ctx.obj` key that tracks whether that flag was provided. Normally that flag just affects `ctx.obj["selected_environment"]` . It was previously considered, but deemed low-priority since running the compile command manually seems like it'll be a rare occurrence.

I was surprised by the fact that `--no-environment` didn't produce `meltano-manifest.json` (i.e. the no environment manifest) earlier today during testing, and _I was the one who implemented it_, so that's reason enough for me to update this.

Reply:
Resolved in ecc9fe4